app/javascript/packages/memorable-date/index.ts

```diff
@@ -0,0 +1,377 @@
+    const day = this.dayInput;
+    const year = this.yearInput;
+
+    return [month, day, year].filter(Boolean) as HTMLInputElement[];
+  }
+
+  /**
+   * The configured minimum valid value for the date, based on "min" HTML attribute
+   */
+  get min(): Date | null {
+    return this.getDateAttribute('min');
+  }
+
+  /**
+   * The configured maximum valid value for the date, based on "max" HTML attribute
+   */
+  get max(): Date | null {
+    return this.getDateAttribute('max');
+  }
+
+  connectedCallback() {
+    const { allInputs } = this;
+
+    const inputListener = (event: Event) => {
+      this.validate();
+
+      // Artificially trigger input events on all inputs
+      // for input events on one input. This makes the corresponding
+      // <lg-validated-field> elements remove error styling from all
+      // memorable-date fields at the same time as it hides the error
+      // message (instead of only the selected field).
+      const otherInputs = allInputs.filter((input) => input !== event.target);
+      try {
+        this.removeEventListener('input', inputListener);
+        otherInputs.forEach((input) => {
+          // Prevent recursion by removing listener temporarily
+          input.dispatchEvent(new CustomEvent('input', { bubbles: true }));
+        });
+      } finally {
+        this.addEventListener('input', inputListener);
+      }
+    };
+
+    this.addEventListener('input', inputListener);
+    this.addEventListener('invalid', () => this.validate(), true);
+  }
+
+  validate(): void {
+    const month = this.monthInput;
+    const day = this.dayInput;
+    const year = this.yearInput;
+
+    if (!(month && day && year)) {
+      // Cannot accurately run validation w/o all fields
+      return;
+    }
+
+    const { errorMessages, rangeErrors } = this.getErrorMessageMappings();
+
+    const hasMissingValues = (
+      [
+        [MemorableDateErrorMessage.MISSING_MONTH_DAY_YEAR, [month, day, year]],
+        [MemorableDateErrorMessage.MISSING_MONTH_DAY, [month, day]],
+        [MemorableDateErrorMessage.MISSING_MONTH_YEAR, [month, year]],
+        [MemorableDateErrorMessage.MISSING_MONTH, [month]],
+        [MemorableDateErrorMessage.MISSING_DAY_YEAR, [day, year]],
+        [MemorableDateErrorMessage.MISSING_DAY, [day]],
+        [MemorableDateErrorMessage.MISSING_YEAR, [year]],
+      ] as ErrorMessageFieldMapping[]
+    ).some(this.checkMissingValues(errorMessages));
+
+    if (hasMissingValues) {
+      return;
+    }
+
+    const hasInvalidValues = (
+      [
+        [MemorableDateErrorMessage.INVALID_MONTH, [month]],
+        [MemorableDateErrorMessage.INVALID_DAY, [day]],
+        [MemorableDateErrorMessage.INVALID_YEAR, [year]],
+      ] as ErrorMessageFieldMapping[]
+    ).some(this.checkFieldsInvalid(errorMessages));
+
+    if (hasInvalidValues) {
+      return;
+    }
+
+    let parsedDate: Date | undefined;
+    try {
+      parsedDate = new Date(`${year.value}-${month.value}-${day.value}`);
```

Comment:
When testing in Safari, I noticed I was getting a "This is not a valid date" error when entering `1/1/2000`. I suspect it has something to do with the month and day values being single digits, where this would try to parse (unsuccessfully) `new Date('2000-1-1')`.

Chrome seems to handle it gracefully.

Maybe we can format the numbers to pad the `0`.

```suggestion
      parsedDate = new Date(`${year.value}-${month.value.padStart(2, '0')}-${day.value.padStart(2, '0')}`);
```